test/handshake.test.ts

```diff
@@ -45,286 +46,273 @@ test('Basic Socket', { timeout: 5000 }, async t => {
 });
 
-test('Socket Double Disconnection', { timeout: 5000 }, async t => {
-	t.plan(2);
-	const [nodeServer, nodeSocket] = await setup(t, ++port);
+test('Socket Double Disconnection', async (done) => {
+	expect.assertions(2);
+	const [nodeServer, nodeSocket] = await setup(done, ++port);
 	const server = nodeSocket.get('Server')!;
 
 	try {
-		t.true(server.disconnect(), 'Successful disconnections should return true.');
-		t.false(server.disconnect(), 'A repeated disconnection should return false.');
+		expect(server.disconnect()).toBeTruthy();
+		expect(server.disconnect()).toBeFalsy();
 	} catch (error) {
-		t.error(error, 'Disconnections from NodeSocket should never throw an error.');
+		expect(error).toBeFalsy();
 	}
 
 	await nodeServer.close();
 });
 
-test('Server Double Disconnection', { timeout: 5000 }, async t => {
-	t.plan(2);
-	const [nodeServer, nodeSocket] = await setup(t, ++port);
+test('Server Double Disconnection', async (done) => {
+	expect.assertions(2);
+	const [nodeServer, nodeSocket] = await setup(done, ++port);
 	const server = nodeServer;
 
 	try {
-		t.true(await server.close(), 'Successful disconnections should return true.');
-		t.false(await server.close(), 'A repeated disconnection should return false.');
+		expect(await server.close()).toBeTruthy();
+		expect(await server.close()).toBeFalsy();
 	} catch (error) {
-		t.error(error, 'Disconnections from NodeSocket should never throw an error.');
+		expect(error).toBeFalsy();
 	}
 
 	try {
 		nodeSocket.disconnectFrom('Server');
 	} catch (error) {
-		t.error(error, 'Disconnection should not error.');
+		expect(error).toBeFalsy();
 	}
 });
 
-test('Socket Connection Retries', { timeout: 7500 }, async t => {
-	t.plan(4);
-	const [nodeServer, nodeSocket] = await setup(t, ++port, { maximumRetries: 3, retryTime: 0 });
+test('Socket Connection Retries', async (done) => {
+	expect.assertions(4);
+	const [nodeServer, nodeSocket] = await setup(done, ++port, { maximumRetries: 3, retryTime: 0 });
 	await nodeServer.close();
 
 	let attempts = nodeSocket.maximumRetries;
 	nodeSocket.on('connecting', () => {
-		t.true(--attempts >= 0, 'This should reconnect exactly 3 times.');
-	});
-	nodeSocket.on('disconnect', () => {
-		t.pass('The client successfully disconnected.');
+		expect(--attempts >= 0).toBeTruthy();
 	});
+	nodeSocket.on('disconnect', () => undefined);
 });
 
-test('Socket Connection No Retries', { timeout: 5000 }, async t => {
-	t.plan(1);
-	const [nodeServer, nodeSocket] = await setup(t, ++port, { maximumRetries: 0 });
+test('Socket Connection No Retries', async (done) => {
+	expect.assertions(1);
+	const [nodeServer, nodeSocket] = await setup(done, ++port, { maximumRetries: 0 });
 	await nodeServer.close();
 
 	nodeSocket.on('connecting', () => {
-		t.fail('The socket should not try to connect.');
-	});
-	nodeSocket.on('disconnect', () => {
-		t.pass('The client successfully disconnected.');
+		done.fail('The socket should not try to connect.');
```

Comment:
done.fail is not a function. I said that in previous review as well. Remove it. It's just `done()` though in most cases you shouldn't even need it.